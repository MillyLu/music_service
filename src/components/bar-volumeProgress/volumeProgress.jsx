import * as Styled from './styles';


export function VolumeProgress({volume, setVolume}){
    return(
        <Styled.VolumeProgress>
            <Styled.VolumeProgressLine data-testid="progress" value={volume} name='range' min={0} max={100} onChange={(e) => setVolume(e.target.value)} />
        </Styled.VolumeProgress>
    )
}