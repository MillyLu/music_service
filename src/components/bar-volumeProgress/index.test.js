/* eslint-disable import/extensions */
/* eslint-disable arrow-body-style */
/* eslint-disable react/function-component-definition */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-undef */
 // import * as ShallowRenderer from 'react-shallow-renderer';
 import { screen, render,  fireEvent } from "@testing-library/react";
// eslint-disable-next-line import/no-unresolved

import { VolumeProgress } from "./volumeProgress";
// import * as Styled from './styles';



describe("<VolumeProgress />", () => {
  describe("Snapshots tests", () => {
    it("should render successfully and match snapshot", () => {
      const { container } = render(<VolumeProgress></VolumeProgress>);

      expect(container).toMatchSnapshot();
    });
  });
});


const setup = () => {
    const setVolume = jest.fn();
    const utils = render(<VolumeProgress volume={60} setVolume={setVolume} />)

    const input = screen.getByTestId("progress")
    return {
      input,
      ...utils,
    }
  }

  test('It should take value 60', () => {

    const {input} = setup()
    fireEvent.change(input, {target: {value: '60'}})

    expect(input.value).toBe('60')
  })


