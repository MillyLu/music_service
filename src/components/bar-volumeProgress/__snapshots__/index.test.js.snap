// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolumeProgress /> Snapshots tests should render successfully and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hBgvgi"
  >
    <input
      class="sc-gswNZR izocci"
      data-testid="progress"
      max="100"
      min="0"
      name="range"
      type="range"
      value=""
    />
  </div>
</div>
`;
